scalafmt-interfaces/src/main/java/org/scalafmt/interfaces/RepositoryCredential.java

```diff
@@ -0,0 +1,24 @@
+package org.scalafmt.interfaces;
+
+public final class RepositoryCredential {
+    public final String host;
+    public final String username;
+    public final String password;
+    public final String realm;
```

Comment:
it's not used in the current implementation, because we pass before the connection. if we figure out how to pass all of credentials, possibly if coursier changes the interface and allows passing them via API rather than `coursier.credentials` java system property, that will change.

is adding a realm later going to be a breaking binary change?

Reply:
With a default value it shouldn't.

Reply:
in that case, removing realm for now.